src/components/Navbar.jsx

```diff
@@ -0,0 +1,114 @@
+// src/components/Navbar.js
+import { useState } from 'react';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+
+const Navbar = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const router = useRouter();
+
+  const getLinkClass = (path) => {
+    const isActive = router.pathname === path;
+    return isActive
+      ? 'py-4 px-2 text-indigo-500 border-b-4 border-indigo-500 font-semibold'
+      : 'py-4 px-2 text-gray-500 border-b-4 border-transparent hover:border-indigo-500 transition duration-300';
+};
+
+  return (
+    <nav className="bg-black shadow-lg">
+      <div className="max-w-6xl mx-auto px-4">
+        <div className="flex justify-between">
+          <div className="flex space-x-7">
+            <div>
+              <Link href="/">
+                <a className="flex items-center py-4 px-2">
+                  <span className="font-semibold text-gray-500 text-lg">RK</span>
+                </a>
+              </Link>
+            </div>
+            <div className="hidden md:flex items-center space-x-1">
+              <Link href="/">
+                <a className={getLinkClass('/')}>Home</a>
+              </Link>
+              <Link href="/about">
+                <a className={getLinkClass('/about')}>About</a>
+              </Link>
+              <Link href="/services">
+                <a className={getLinkClass('/services')}>Services</a>
+              </Link>
+              <Link href="/contact">
+                <a className={getLinkClass('/contact')}>Contact</a>
+              </Link>
+            </div>
+          </div>
+          <div className="md:hidden flex items-center">
+            <button
+              className="outline-none mobile-menu-button"
+              onClick={() => setIsOpen(!isOpen)}
```

Comment:
**Issue**: The use of an inline function for the onClick handler in the button element can lead to performance issues. Each re-render of the component will create a new function, causing the child components to re-render unnecessarily. <br> **Fix**: Extract the inline function into a useCallback hook to prevent unnecessary re-renders of child components. <br> **Code Suggestion**: 
 ```
 @@ -45,3 +45,4 @@
 -            onClick={() => setIsOpen(!isOpen)}
 +            onClick={toggleMenu}
 @@ -6,3 +6,5 @@
 +  const toggleMenu = useCallback(() => setIsOpen(prev => !prev), []);
 ```